src/engine/common_components/SoundEffectPlayer.cs

```diff
@@ -189,6 +189,48 @@ public static bool PlaySoundEffect(this ref SoundEffectPlayer soundEffectPlayer,
             return false;
         }
 
+        /// <summary>
+        ///   Stops all sounds that are currently playing
+        /// </summary>
+        /// <param name="soundEffectPlayer">The player that may have sounds</param>
+        /// <param name="immediately">
+        ///   If false, any looping sounds will only stop playing after their current loop ends
+        /// </param>
+        public static void StopAllSounds(this ref SoundEffectPlayer soundEffectPlayer, bool immediately = true)
+        {
+            SoundEffectSlot[]? slots = soundEffectPlayer.SoundEffectSlots;
+
+            if (slots == null)
+                return;
+
+            lock (slots)
+            {
+                int slotCount = slots.Length;
+
+                for (int i = 0; i < slotCount; ++i)
+                {
+                    ref var slot = ref slots[i];
+
+                    if (!slot.Play)
+                        continue;
+
+                    if (slot.Loop && !immediately)
+                    {
+                        // Stop after current loop
+                        slot.Loop = false;
+                    }
+                    else
+                    {
+                        slot.Play = false;
+                    }
+
+                    slot.InternalAppliedState = false;
+                }
+            }
+
+            soundEffectPlayer.SoundsApplied = false;
```

Comment:
It would be better to have this inside the lock to be consistent with other methods and ensure everyone who uses lock on the slots will see this updated variable at the same time.